lib/palindrome_check.rb

```diff
@@ -1,5 +1,37 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
+
+  return false if my_phrase == nil
+
+  i = 0
+  j = my_phrase.length - 1
+
+  while i < j
+    while my_phrase[i] == ' '
```

Comment:
In the two nested inner loops, continue to check and confirm that i is less than j before comparing the character at the index i (or j)  with a white space.